# 安装配置ChatTTS环境-推荐

sb 2024-6-11

## 删除原有的ChatTTS目录

In [1]:
!rm -rf ChatTTS

## 从Github下载ChatTTS

In [2]:
!git clone https://github.com/2noise/ChatTTS.git

Cloning into 'ChatTTS'...
remote: Enumerating objects: 2614, done.
remote: Counting objects: 100% (739/739), done.
remote: Compressing objects: 100% (331/331), done.
remote: Total 2614 (delta 477), reused 453 (delta 398), pack-reused 1875 (from 1)
Receiving objects: 100% (2614/2614), 7.98 MiB | 18.62 MiB/s, done.
Resolving deltas: 100% (1578/1578), done.
Updating files: 100% (93/93), done.


In [3]:
# 切换到指定版本
# !git reset --hard 0359f28545c7285d7ec3d98ca25c1099e7890c4a

## 进入ChatTTS主目录

In [4]:
import os

In [5]:
os.chdir('ChatTTS')

## 安装所需工具包

In [6]:
!pip install -r requirements.txt

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.6/57.6 kB 722.9 kB/s eta 0:00:00a 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.1/44.1 kB 3.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.5/40.5 kB 1.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.8/60.8 kB 2.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.7/3.7 MB 15.0 MB/s eta 0:00:0000:010:01
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 50.8/50.8 kB 3.3 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.9/89.9 kB 6.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 149.4/149.4 kB 10.5 MB/s eta 0:00:00
INFO: pip is looking at multiple versions of wetextprocessing to determine which version is compatible with other requirements. This could take a while.
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
INFO: pip is

In [7]:
!pip install soundfile torchaudio pynini==2.1.5 nemo_text_processing gradio

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 4.4 MB/s eta 0:00:00a 0:00:010m

[notice] A new release of pip is available: 23.3.1 -> 24.3.1
[notice] To update, run: python -m pip install --upgrade pip


## 验证安装成功

In [8]:
import torchaudio
import torch
import soundfile
from IPython.display import Audio

In [9]:
import ChatTTS

如果导入上述工具包没有报错，则说明安装配置成功！

## 修改代码

将`ChatTTS/core.py`的143行到149行注释掉，效果如下：

```python
        # if do_text_normalization:
        #     for i, t in enumerate(text):
        #         _lang = detect_language(t) if lang is None else lang
        #         self.init_normalizer(_lang)
        #         text[i] = self.normalizer[_lang](t)
        #         if _lang == 'zh':
        #             text[i] = apply_half2full_map(text[i])
```

## 从Hugging Face下载预训练模型

In [10]:
chat = ChatTTS.Chat()

In [11]:
chat.load(compile=False) # Set to True for better performance

]0;RVC Models Downloader

time="2024-11-14T06:47:00Z" level=info msg="RVC Models Downloader start at 2024-11-14 06:47:00 (UTC)"
time="2024-11-14T06:47:00Z" level=info msg="operating system: linux, architecture: amd64"
time="2024-11-14T06:47:01Z" level=info msg="is ipv6 available: false"
time="2024-11-14T06:47:01Z" level=info msg="#1 open target folder 'asset'."
time="2024-11-14T06:47:01Z" level=info msg="#1 download copy: '[asset/Decoder.safetensors asset/DVAE.safetensors asset/Embed.safetensors asset/Vocos.safetensors]'."
time="2024-11-14T06:47:01Z" level=info msg="#1.4 get: https://huggingface.co/2Noise/ChatTTS/resolve/main/asset/Vocos.safetensors"
time="2024-11-14T06:47:01Z" level=info msg="#1.1 get: https://huggingface.co/2Noise/ChatTTS/resolve/main/asset/Decoder.safetensors"
time="2024-11-14T06:47:01Z" level=info msg="#1.2 get: https://huggingface.co/2Noise/ChatTTS/resolve/main/asset/DVAE.safetensors"
time="2024-11-14T06:47:01Z" level=info msg="#1.3 get: https://huggingface.co/2Noise/ChatTTS/resolve/main/a

True

如果无法下载，卡着不动，请移步至B2代码教程